### PR TITLE
Simple drag and drop

### PR DESCRIPTION
**Pull request by Markvanderuit**
We now have simple blocks that are draggable and stretchable in all directions, in a very large
scrollable area that can also handle resizes. Next up we'll put them in a grid of some sort.

### CODE REVIEW

File: src/main/java/gui/TimetableBlock.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+     * Constructor class for timetable block.
+     * @param pane parent area this block is located in.
+     */
+    TimetableBlock(RootCenterArea pane) {
+        // some basic temporary styling for the block. That's size, position, and style.
+        setLayoutX(100.0);
+        setLayoutY(100.0);
+        setMinHeight(100.0);
+        setMinWidth(100.0);
+        setStyle(normalStyle);
```
Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java:90]: Could these settings be tweakable? Like make them fields with getters and setters, maybe even initialize them in the constructor

File: src/main/java/gui/RootCenterArea.java [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+        TimetableBlock rect2 = new TimetableBlock(this);
+        TimetableBlock rect3 = new TimetableBlock(this);
+        TimetableBlock rect4 = new TimetableBlock(this);
 
-    RootCenterArea() {
-        // Dummy class for now.
+        contentPane.getChildren().add(rect1);
+        contentPane.getChildren().add(rect2);
+        contentPane.getChildren().add(rect3);
+        contentPane.getChildren().add(rect4);
```
Comment (martijnjanssen) [inline src/main/java/gui/RootCenterArea.java:40]: Is there no `addAll()` method that can be used here?

File: src/main/java/gui/TimetableBlock.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            }
+        });
+    }
+
+    /**
+     * Helper function for MouseDragged event.
+     * @param pane pane this block is located in.
+     * @param event the mousedrag event in question.
+     */
+    private void onMouseDraggedHelper(RootCenterArea pane, MouseEvent event) {
```
Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java]: You could also return an eventhandler with the body of this method as the body of the handle method here. That would clean up the code in the constructor

File: src/main/java/gui/TimetableBlock.java (hunk 1 above)
Comment (martijnjanssen) [inline src/main/java/gui/TimetableBlock.java:90]: I believe these are just settings for testing, not final values. Am I correct @Ezlo123 ?

Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java:90]: But in the final versions, x, y, width and height values must be settable, therefore i suggest to put them in fields. The testing values of 100.0 can remain for now.

Comment (Markvanderuit) [inline src/main/java/gui/TimetableBlock.java:90]: Just testing settings. Did, however, add 2 constructors that allow you to specifiy coordinates and width, if necessary.

File: src/main/java/gui/RootCenterArea.java (hunk 2 above)
Comment (Markvanderuit) [inline src/main/java/gui/RootCenterArea.java:40]: Sure, but these are just for testing. Seems unnecessary when they'll be gone tomorrow.

**Comment by martijnjanssen** [comment]
Right now, when dragging the bottom edge of a `TimetableBlock` the edge stops at the upper bound. When doing this to the top edge, the entire block slides. Is this fixable?

Could we also specify a minimum height for the element?

File: src/main/java/gui/TimetableBlock.java (hunk 3 above)
Comment (Markvanderuit) [inline src/main/java/gui/TimetableBlock.java]: Could you clarify this a bit more?

**Comment by martijnjanssen** [comment]
The mouse also changes in position relative to the block when exiting the screen when dragging the block. Can we remove this? It would be better to have the mouse reset to the initial position, right?

File: src/main/java/gui/TimetableBlock.java (hunk 3 above)
Comment (bart-de-jonge) [inline src/main/java/gui/TimetableBlock.java]: In the constructor you now make new eventhandler and call this method in the body, why not return an actual eventhandler from this function and add it like: `setOnMouseReleased(onMouseDraggedHelper())` seems a lot cleaner to me

**Comment by Markvanderuit** [comment]
@martijn9612 I'm fixing the first dragging issue right now. However, the position change relative to the block (when your mouse exits the screen) would probably mean forcing the mouse to move. That's one of the things you should, from a user's point of view, probably never do. Not to mention that I honestly don't know if the OS actually lets me do that. The reason you see it move is because the block encounters an edge (the edge I have specified). What we could do instead, is make the view move with the mouse when dragging.

**Comment by martijnjanssen** [comment]
We'll discuss this later, I can't really explain what I mean over git. We are meeting again soon!

**Comment by bart-de-jonge** [comment]
Great work fixing those bugs

**Pull request merged**